app/src/features/datasets/components/DatasetArtifacts.tsx

```diff
@@ -269,6 +190,20 @@ const DatasetAttachments: React.FC<IDatasetAttachmentsProps> = (props) => {
             slots={{
               noRowsOverlay: NoArtifactRowsOverlay
             }}
+            onRowSelectionModelChange={(params) => {
```

Comment:
If we end up just maintain the selected artifact IDs, we should be able to avoid this callback altogether:

```typescript
  onRowSelectionModelChange={setSelectedArtifactIds}
/>
```